tfx/extensions/google_cloud_kubernetes/runner.py

```diff
@@ -0,0 +1,245 @@
+                    port=5000,
+                ),
+            ],
+        ),
+    )
+    try:
+      api_instance.create_namespaced_service(namespace='default', body=service)
+    except ApiException as e:
+      logging.error(
+          'Exception when calling CoreV1Api->create_namespaced_service: %s' % e)
+  logging.info('created {} worker services'.format(num_workers))
+
+
+def delete_worker_services(training_inputs: Dict[Text, Any],
+                           unique_id: Text):
+  """Clean up worker services deployed to the kubernetes cluster."""
+  num_workers = training_inputs.get('num_workers', 1)
+  service_names = _build_service_names(num_workers=num_workers,
+                                       unique_id=unique_id)
+  api_instance = kube_utils.make_core_v1_api()
+  for service_name in service_names:
+    try:
+      api_instance.delete_namespaced_service(namespace='default',
+                                             name=service_name)
+    except ApiException as e:
+      logging.error(
+          'Exception when calling CoreV1Api->delete_namespaced_service: %s' % e)
+  logging.info('Deleted {} worker services'.format(num_workers))
+
+
+def start_gke_training(input_dict: Dict[Text, List[types.Artifact]],
+                       output_dict: Dict[Text, List[types.Artifact]],
+                       exec_properties: Dict[Text,
+                                             Any], executor_class_path: Text,
+                       training_inputs: Dict[Text,
+                                             Any], unique_id: Text):
+  """Start a trainer job on Google Kubernetes Engine (GKE).
+
+  This is done by forwarding the inputs/outputs/exec_properties to the
+  tfx.scripts.run_executor module on a AI Platform training job interpreter.
+
+  Args:
+    input_dict: Passthrough input dict for tfx.components.Trainer.executor.
+    output_dict: Passthrough input dict for tfx.components.Trainer.executor.
+    exec_properties: Passthrough input dict for tfx.components.Trainer.executor.
+    executor_class_path: class path for TFX core default trainer.
+    training_inputs: Training input argument for GKE.
+      'num_workers', 'num_gpus_per_worker' and 'tfx_image' will be consumed.
+
+  Returns:
+    None
+  Raises:
+    RuntimeError: if the Google Kubernetes Engine training job failed/cancelled.
+  """
+  training_inputs = training_inputs.copy()
+
+  json_inputs = artifact_utils.jsonify_artifact_dict(input_dict)
+  logging.info('json_inputs=\'%s\'.', json_inputs)
+  json_outputs = artifact_utils.jsonify_artifact_dict(output_dict)
+  logging.info('json_outputs=\'%s\'.', json_outputs)
+  json_exec_properties = json.dumps(exec_properties, sort_keys=True)
+  logging.info('json_exec_properties=\'%s\'.', json_exec_properties)
+
+
+  # We use custom containers to launch training on GKE, which invokes
+  # the specified image using the container's entrypoint. The default
+  # entrypoint for TFX containers is to call scripts/run_executor.py. The
```

Comment:
Is this the default entrypoint?  I think it's just the entrypoint we use here in the Kubernetes trainer code, right?

Reply:
Right, I corrected the comments.